.github/workflows/build-dev-image.yml

```
@@ -49,3 +51,51 @@ jobs:
+            echo "WATCHTOWER_API_URL / WATCHTOWER_API_TOKEN not configured."
+            echo "Container will pick up the new image on Watchtower's next 60-second poll."
+            echo "To enable direct trigger: set WATCHTOWER_API_URL and WATCHTOWER_API_TOKEN in repo secrets."
+          fi
+
+      # Post a notification to #infra Discord channel after a successful image push.
+      # Requires DISCORD_INFRA_WEBHOOK_URL secret to be set.
+      - name: Notify Discord infra channel
+        if: success()
+        env:
+          DISCORD_INFRA_WEBHOOK_URL: ${{ secrets.DISCORD_INFRA_WEBHOOK_URL }}
+        run: |
```

actionlint (1.7.12)

[error] "github.event.head_commit.message" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/reference/security/secure-use#good-practices-for-mitigating-script-injection-attacks for more details

(expression)